src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,143 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int listLength;
+
     @Override
     public void add(T value) {
-
+        if (listLength < array.length) {
+            array[listLength] = value;
+            listLength++;
+        } else {
+            array = grow(array);
+            array[listLength] = value;
+            listLength++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < listLength && index >= 0) {
+            if (listLength == array.length) {
+                array = grow(array);
+            }
```

Comment:
combine this check with simple addition of elem

Reply:
didn't understand this note. why do I need use simple addition of elem, if I have to add an element by index?